### PR TITLE
Document custom component paths

### LINKED ISSUE #298: componentPaths cannot have a folder with a `index.js` in it
### What's up?

I happened to put one folder with `index.js` in `componentPaths`. The `cosmos.config.js` is like below.

```
module.exports = {
  componentPaths: [
      'src/components/Home'
  ]
};
```

The `src/components/Home` folder has one `index.js` file in it. After launching cosmos, it fails with error message 
>ERROR in ./~/react-cosmos-webpack/lib/user-modules.js
Module build failed: Error: Must implement getComponentName when using exact file paths in componentPaths

I dug into the code and found [this line](https://github.com/react-cosmos/react-cosmos/blob/master/packages/react-cosmos-voyager/src/index.js#L56) throws the Error. It turns out that [`react-cosmos-config`](https://github.com/react-cosmos/react-cosmos/blob/master/packages/react-cosmos-config/src/index.js#L7) utilizes `resolveFrom` to normalize component path. The npm `resolveFrom` returns path with `index.js` if there is `index.js` in given folder, which make `react-cosmos-voyager` think it is exact file paths.

IMHO, `componentPaths` should allow existence of `index.js` without requiring `getComponentName`.

### Mkay, tell me more...

There is no documentation about `getComponentName` found.

### PR DESCRIPTION
**Pull request by skidding**
Fixes #298

Added a separate section instead of putting `getComponentName` with all the other options because I'm planning to split all options per use case sections in the future.

### CODE REVIEW

**Comment by skidding** [comment]
@mocheng please review

File: README.md [hunk 1]
```diff
@@ -160,6 +160,29 @@ module.exports = {
 };
 ```
 
+#### Custom component paths
+
+The `componentPaths` option supports both dir and file paths. Most of the times using a dir path is enough (e.g. "src/components"), but we might need to target the exact file paths when component files sit next non-component files. Here's an example for this type of setup:
```
Comment (ovidiubute) [inline README.md]: > when component files sit next non-component files

Should be: "when component files sit next **to** non-component files"

Comment (skidding) [inline README.md]: Fixed!

**Comment by mocheng** [comment]
+1

**Pull request merged**